packages/smooth_app/lib/pages/scan/ml_kit_scan_page.dart

```diff
@@ -129,32 +129,36 @@ class MLKitScannerPageState extends State<MLKitScannerPage> {
     _controller?.setFocusMode(FocusMode.auto);
     _controller?.lockCaptureOrientation(DeviceOrientation.portraitUp);
 
-    _controller = _controller;
-
-    try {
-      await _controller!.initialize();
+    // If the controller is initialized update the UI.
+    _controller?.addListener(() {
       if (mounted) {
         setState(() {});
       }
       if (_controller!.value.hasError) {
-        debugPrint(_controller!.value.errorDescription);
+        // TODO(M123): Handle errors better
+        debugPrint(_controller?.value.errorDescription);
```

Comment:
That's another inconsistent pattern if was referring to: `!` then `?`.
If you have good reasons to think it's not null, ok for `!` in both places, but not a mix that is confusing for the code reviewer.